swarm/storage/mru/resource.go

```diff
@@ -146,115 +161,61 @@ func (r *resource) Name() string {
-// and update chunk will always have a value > 0 there.)
-//
-// The root entry tells the requester from when the mutable resource was
-// first added (block number) and in which block number to look for the
-// actual updates. Thus, a resource update for identifier "føø.bar"
-// starting at block 4200 with frequency 42 will have updates on block 4242,
-// 4284, 4326 and so on.
-//
-// Actual data updates are also made in the form of swarm chunks. The keys
-// of the updates are the hash of a concatenation of properties as follows:
-//
-// sha256(period|version|namehash)
-//
-// The period is (currentblock - startblock) / frequency
-//
-// Using our previous example, this means that a period 3 will have 4326 as
-// the block number.
-//
-// If more than one update is made to the same block number, incremental
-// version numbers are used successively.
-//
-// A lookup agent need only know the identifier name in order to get the versions
-//
-// the resourcedata is:
-// headerlength|period|version|identifier|data
-//
-// if a validator is active, the chunk data is:
-// resourcedata|sign(resourcedata)
-// otherwise, the chunk data is the same as the resourcedata
-//
-// headerlength is a 16 bit value containing the byte length of period|version|name
-//
-// TODO: Include modtime in chunk data + signature
+// Handler is the API for Mutable Resources
+// It enables creating, updating, syncing and retrieving resources and their update data
 type Handler struct {
 	chunkStore      *storage.NetStore
 	HashSize        int
 	signer          Signer
 	headerGetter    headerGetter
-	ownerValidator  ownerValidator
 	resources       map[string]*resource
 	hashPool        sync.Pool
 	resourceLock    sync.RWMutex
 	storeTimeout    time.Duration
-	queryMaxPeriods *LookupParams
+	queryMaxPeriods uint32
 }
 
+// HandlerParams pass parameters to the Handler constructor NewHandler
+// Signer and HeaderGetter are mandatory parameters
 type HandlerParams struct {
-	QueryMaxPeriods *LookupParams
+	QueryMaxPeriods uint32
 	Signer          Signer
 	HeaderGetter    headerGetter
-	OwnerValidator  ownerValidator
 }
 
-// Create or open resource update chunk store
+// NewHandler creates a new Mutable Resource API
 func NewHandler(params *HandlerParams) (*Handler, error) {
-	if params.QueryMaxPeriods == nil {
-		params.QueryMaxPeriods = &LookupParams{
-			Limit: false,
-		}
+	if params.Signer == nil {
+		return nil, NewError(ErrInit, "Signer cannot be nil")
+	}
+	if params.HeaderGetter == nil {
+		return nil, NewError(ErrInit, "HeaderGetter cannot be nil")
 	}
 	rh := &Handler{
-		headerGetter:   params.HeaderGetter,
-		ownerValidator: params.OwnerValidator,
-		resources:      make(map[string]*resource),
-		storeTimeout:   defaultStoreTimeout,
-		signer:         params.Signer,
+		headerGetter: params.HeaderGetter,
+		resources:    make(map[string]*resource),
+		storeTimeout: defaultStoreTimeout,
```

Comment:
why set on handler if value can only be default?

Reply:
might want to set it per instantiation in the future.